### PR TITLE
fetch restricted runs in publisher

### PR DESCRIPTION
**Pull request by zawan-ila**
[PROD-4088](https://2u-internal.atlassian.net/browse/PROD-4088)
-------
This PR adds adds the restriction params to the API calls in Publisher to ensure that restricted runs are always fetched/updated properly in Publisher.

## Testing

1. Check out this branch
2. Open/Create an executive education course in publisher
3. Add a restriction type. Then change it. Finally remove it. 
4. Verify that the behavior is as expected.

### CODE REVIEW

**Review by AfaqShuaib09 — APPROVED** [review]
Looks good to me, better to have a unittest for this usecase

**Comment by zawan-ila** [comment]
> Looks good to me, better to have a unittest for this usecase

I don't see many tests for `discoveryApiService`. I could add a test that asserts that the include_restricted param is included in the url, but I am not sure how helpful that would be. If you think that would be good enough, I am happy to add it. Or if you have ideas for a different test, please let me know about that too.

File: src/data/services/DiscoveryDataApiService.js [hunk 1]
```diff
@@ -10,6 +11,7 @@ class DiscoveryDataApiService {
     const queryParams = {
       editable: 1,
       exclude_utm: 1,
+      include_restricted: restrictionTypeValues,
```
Comment (DawoudSheraz) [inline src/data/services/DiscoveryDataApiService.js:14]: hmmm, do we not need any test updates for this?

File: src/data/constants/index.js [hunk 2]
```diff
@@ -33,6 +33,14 @@ const COURSE_URL_SLUG_VALIDATION_MESSAGE = {
   [COURSE_URL_SLUG_PATTERN]: 'Course URL slug contains lowercase letters, numbers, underscores, and dashes only and must be in the format <custom-url-slug> or learn/<primary_subject>/<org-slug>-<course_slug>.',
 };
 
+const RESTRICTION_TYPE_OPTIONS = [
+  { value: '', label: '--------' },
+  { value: 'custom-b2b-enterprise', label: 'Custom Enterprise' },
+  { value: 'custom-b2c', label: 'Custom B2C' },
+];
+
+const RESTRICTION_TYPE_VALUES = RESTRICTION_TYPE_OPTIONS.map(opt => opt.value).filter(Boolean).join(',');
```
Comment (DawoudSheraz) [inline src/data/constants/index.js:42]: 👍🏽 on that Boolean filter

**Review by DawoudSheraz — APPROVED** [review]

File: src/data/services/DiscoveryDataApiService.js (hunk 1 above)
Comment (zawan-ila) [reply src/data/services/DiscoveryDataApiService.js:14]: No. There are virtually no tests for this file.

**Pull request merged**